src/client.py

```diff
@@ -0,0 +1,88 @@
+from numbers import Number
+
+from open_feature import OpenFeature
+from src.flag_type import FlagType
+
+
+class OpenFeatureClient:
+    def __init__(self, name: str, version: str, context, hooks: list = []):
+        self.name = name
+        self.version = version
+        self.context = context
+        self.hooks = hooks
+
+    def get_boolean_value(
+        self, key: str, defaultValue: bool, evaluationContext, flagEvaluationOptions
+    ) -> bool:
+        return self.evaluate_flag(
+            FlagType.BOOLEAN,
+            key,
+            defaultValue,
+            evaluationContext,
+            flagEvaluationOptions,
+        )
+
+    def get_boolean_details(
+        self, key: str, defaultValue: bool, evaluationContext, flagEvaluationOptions
+    ):
+        return self.get_boolean_details(
```

Comment:
Won't this just end up in an infinite loop?

Reply:
Agreed it should be calling the provider method. This is now fixed